src/middlewared/middlewared/plugins/disk_/sync.py

```diff
@@ -17,11 +17,10 @@ async def sync(self, name):
         """
         Syncs a disk `name` with the database cache.
         """
-        if (
-            not await self.middleware.call('system.is_freenas') and
-            await self.middleware.call('failover.licensed') and
-            await self.middleware.call('failover.status') == 'BACKUP'
-        ):
+        is_enterprise = await self.middleware.call('system.is_enterprise')
+        is_ha = await self.middleware.call('failover.licensed')
+        is_backup = await self.middleware.call('failover.status') == 'BACKUP'
+        if is_enterprise and is_ha and is_backup:
```

Comment:
You are trying to get failover status even on non-enterprise systems or systems without failover licensed

Reply:
It should be fine, we return SINGLE on that case

Reply:
@william-gr but that takes an `interface.query`

Maybe we can use this chance to make `failover.status` immediately return `SINGLE` if we are not enterprise or failover is not licensed? And simplify many logic expressions in this diff.

Reply:
Since code base is merged, there is no reason to make 3 calls here and I can simplify this down to 1 call.

Reply:
Agreed

Reply:
@themylogin I like your idea, however, after looking at this a little more closely it will cause further complications for certain scenarios. For example, if we don't want to run code on the standby node, we have a check like:
`if failover.status == 'BACKUP': return` which will now have to become `failover.status not in ('MASTER', 'SINGLE'): return`. I'll change this PR to only replace the `is_freenas` calls and I'll open a separate ticket about unifying the `failover` checks in general since it needs to be cleaned up now that all the code is merged into 1 repo.

Reply:
@yocalebo Sorry, I don't understand how `if failover.status == 'BACKUP': return` will change its behavior if we still will be returning `SINGLE`, just doing it faster.

Reply:
I'm going to keep the changes in this PR to what it was originally set out to do. Simply replace `is_freenas` with `is_enterprise`. I'll open a separate PR for condensing the `failover.*` api calls.